src/recs.js

```diff
@@ -71,11 +78,17 @@ export async function initTemplate(options) {
 	const description = answers && answers.description;
 	const templateDir = (answers && answers.directory) || templateDefaultDir;
 	const componentName = pascalCase(name);
-
 	try {
-		await writeTemplateSettings(path.resolve(process.cwd(), templateDir, `${componentName}.json`), generateTemplateSettings({ name, description }));
+		await writeTemplateSettings(
+			path.resolve(process.cwd(), templateDir, `${componentName}.json`),
+			generateTemplateSettings({ name, description, type: answers.type })
+		);
 		if (framework) {
-			await writeTemplateSettings(path.resolve(process.cwd(), templateDir, `${componentName}.jsx`), framework.template.component(componentName));
+			if (answers.type === 'email') {
```

Comment:
If previously suggestion mapping restructure is used this could be changed to a one liner and would work for any new template types created in the future.